core/src/main/java/com/arcrobotics/ftclib/hardware/motors/MotorGroup.java

```diff
@@ -40,6 +48,113 @@ public double get() {
         return group[0].get();
     }
 
+    /**
+     * @return All motor target speeds as a percentage of output
+     */
+    public List<Double> getSpeeds() {
+        return Arrays.stream(group)
+                .map(Motor::get)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public double getVelocity() {
+        return group[0].getCorrectedVelocity();
```

Comment:
So, this concept is that of a "leader-followers." Only the lead motor is actually doing any calculations. Every other motor has their speed set to whatever the first one's is set to, so that they run together at the same speed always (in theory). `getCorrectedVelocity()` shouldn't break on non-through bore because it checks to see if the velocity and real velocity are super far apart, meaning something went wrong. If nothing went wrong, nothing happens.

Reply:
See https://github.com/FTCLib/FTCLib/blob/dev/core/src/main/java/com/arcrobotics/ftclib/hardware/motors/Motor.java#L171